File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        i,j,length,sum_ele = 0,0,float('inf'),0
        for j in range(len(nums)):
            sum_ele += nums[j]
            while sum_ele >= target:
                length = min(length,j - i + 1)
                sum_ele -= nums[i]
                i += 1
        return length if length != float('inf') else 0
        

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0,right = 0,sum = 0,len = Integer.MAX_VALUE;
        for(right = 0;right < nums.length;right++){
            sum += nums[right];
            while (sum >= target){
                if((right - left + 1) < len){
                    len = right - left + 1;
                }
                sum -= nums[left];
                left++;
            }
        }
        return len != Integer.MAX_VALUE ? len : 0;
    }
}